### PR TITLE
More Login Tests

### PR DESCRIPTION
**Pull request by brandonpage**
Tests Added:
- React Native Login (iOS and Android)
- React Native Smart Sync Explorer Template  Login (iOS and Android)
- Swift Smart Sync Explorer Template Login 
- Advanced Auth Login (iOS and Android)
- Passcode Creation (iOS and Android)

iOS hybrid_local/hybrid_remote **will fail** due to an issue with packager, which can also be seen here: https://circleci.com/workflow-run/3eda8866-f7c7-416d-82e2-d1cce9eab377.  However, when the issue is fixed in packager repo, the tests here will pass without modification.

### CODE REVIEW

File: Android/app/build.gradle [hunk 1]
```diff
@@ -8,25 +8,26 @@ buildscript {
...
+        classpath 'com.android.tools.build:gradle:3.5.0'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
     }
 }
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 29
     defaultConfig {
         minSdkVersion 21
```
Comment (bhariharan) [inline Android/app/build.gradle]: `minSdkVersion` should be `23`.

File: Android/app/build.gradle [hunk 2]
```diff
@@ -8,25 +8,26 @@ buildscript {
...
-        targetSdkVersion 28
+        targetSdkVersion 29
         versionCode 1
         versionName '1.0'
         testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
     }
+    buildToolsVersion = '29.0.0 rc3'
```
Comment (bhariharan) [inline Android/app/build.gradle]: Should be `29.0.2`.

File: iOS/SalesforceMobileSDK-UITest.xcodeproj/xcuserdata/bpage.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>SchemeUserState</key>
```
Comment (bhariharan) [inline iOS/SalesforceMobileSDK-UITest.xcodeproj/xcuserdata/bpage.xcuserdatad/xcschemes/xcschememanagement.plist]: This should not be checked in. Add `*/xcuserdata` to `.gitignore`.

**Comment by brandonpage** [comment]
FYI All the Android variants (except hybrid local/remote) are failing to compile because the templates are still min api 21 but the sdk is now 23.

**Comment by bhariharan** [comment]
Yeah, I'm fixing the templates right now. Will send a PR today.

**Comment by wmathurin** [comment]
Merging this.
Will send a new PR for xcode 11 / ios 12.4 min / MobileSync changes.

**Pull request merged**